### PR TITLE
Start FPGA getting started guide in target.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Start getting started guide for the FPGA test shield.

### CODE REVIEW

**Comment by AnotherButler** [comment]
ping @mprse

File: docs/porting/target/target.md [hunk 1]
```diff
@@ -85,3 +85,18 @@ The Mbed OS HAL provides a set of conformance tests for the Mbed OS HAL APIs. Yo
 ```
 mbed test -t <toolchain> -m <target> -n mbed-os-tests-mbed_hal*
 ```
+
```
Comment (mprse) [inline docs/porting/target/target.md:88]: if PR https://github.com/ARMmbed/mbed-os/pull/10820 will come in in current form, then:
`mbed-os-tests-mbed_hal*`  -> `tests-mbed_hal_fpga_ci_test_shield*`.

File: docs/porting/target/target.md [hunk 2]
```diff
@@ -85,3 +85,18 @@ The Mbed OS HAL provides a set of conformance tests for the Mbed OS HAL APIs. Yo
...
+To run the FPGA test shield test, perform the following steps:
+
+1. Ensure the FPGA test shield is programmed with the latest firmware version.
+1. Connect the FPGA test shield to the Device Under Test.
+1. Set the jumper on the FPGA test shield to select USB power.
+1. Connect the power to the FPGA test shield using a USB cable.
+1. Add the `FPGA_CI_TEST_SHIELD` component to the Device Under Test in the `targets.json` configuration file.
+1. Run the test:
+
+   `mbed test -t GCC_ARM -m K66F -n tests-mbed_hal_fpga_ci_test_shield-spi_slave -v`
```
Comment (mprse) [inline docs/porting/target/target.md]: At the moment `spi_slave` will be available only on spi feature branch. I suggest selecting different test which will be available on master (e.g. `tests-mbed_hal_fpga_ci_test_shield-spi`).

**Review by mprse — APPROVED** [review]
Only 2 minor comments, otherwise ok.

File: docs/porting/target/target.md (hunk 1 above)
Comment (mprse) [reply docs/porting/target/target.md:88]: I just realized that this line is for general HAL testing, so please ignore this comment.

**Comment by mprse** [comment]
@jamesbeyond ping

**Comment by AnotherButler** [comment]
As this has been approved by an engineer, I'm going to merge it. If there are any other updates, please raise them in a new PR.

**Pull request merged**